runtime/parachains/src/runtime_api_impl/v1.rs

```diff
@@ -190,31 +190,42 @@ fn with_assumption<Config, T, F>(
 pub fn full_validation_data<T: initializer::Config>(
 	para_id: ParaId,
 	assumption: OccupiedCoreAssumption,
-)
-	-> Option<ValidationData<T::BlockNumber>>
-{
+) -> Option<ValidationData<T::BlockNumber>> {
+	use parity_scale_codec::Decode as _;
 	let relay_parent_number = <frame_system::Module<T>>::block_number();
-	with_assumption::<T, _, _>(
-		para_id,
-		assumption,
-		|| Some(ValidationData {
-			persisted: crate::util::make_persisted_validation_data::<T>(para_id, relay_parent_number)?,
-			transient: crate::util::make_transient_validation_data::<T>(para_id, relay_parent_number)?,
-		}),
-	)
+	let relay_storage_root = Hash::decode(&mut &sp_io::storage::root()[..])
```

Comment:
How heavy is this?

Reply:
I believe this depends on the size of the delta. Since that's a fresh runtime call - should be no-op.